### PR TITLE
[feat]: [CDS-47006]: Added Elastigroup swimlane manager-delegate objects

### PR DESCRIPTION
**Pull request by arvind-choudhary-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41492)

### CODE REVIEW

**Review by vitsafronovici — DISMISSED** [review]
> Reviewed 53 of 53 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41492)* status: 53 file(s) to review. 1 of 2 approvals obtained.

File: 970-ng-commons/src/main/java/io/harness/beans/SwaggerConstants.java [hunk 1]
```diff
@@ -36,5 +36,8 @@ public interface SwaggerConstants {
       "io.harness.cdng.elastigroup.ElastigroupConfiguration";
   String INSTANCES_DEFINITION_YAML_ELASTIGROUP_CONFIGURATION_CLASSPATH =
       "io.harness.cdng.elastigroup.ElastigroupInstances";
+
   String FILTERS_MATCHTYPE_ENUM_CLASSPATH = "io.harness.cdng.environment.filters.MatchType";
+  String LOAD_BALANCER_CONFIGURATION_CLASSPATH = "[Lio.harness.cdng.elastigroup.LoadBalancer;";
```
Comment (lovish1999) [inline 970-ng-commons/src/main/java/io/harness/beans/SwaggerConstants.java:41]: is this right ? "[Lio.harness.cdng.elastigroup.LoadBalancer;";

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/spot/elastigroup/deploy/ElastigroupDeployTaskResponse.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@OwnedBy(CDP)
+public class ElastigroupDeployTaskResponse implements DelegateTaskNotifyResponseData {
+  @NonFinal @Setter DelegateMetaInfo delegateMetaInfo;
+  CommandExecutionStatus status;
+  String errorMessage;
+  UnitProgressData unitProgressData;
+
+  List<String> ec2InstanceIdsAdded;
+  List<String> ec2InstanceIdsExisting;
+}
```
[the remark this answers is not in the document]
Comment (arvind-choudhary-h) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/spot/elastigroup/deploy/ElastigroupDeployTaskResponse.java:35]: Yes, we just added independent objects for now. Second PR is on the row.

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java [hunk 3]
```diff
@@ -81,5 +81,9 @@ public interface StepSpecTypeConstants {
   String CHAOS_STEP = "Chaos";
 
   String ELASTIGROUP_SETUP = "ElastigroupSetup";
+
   String GITOPS_FETCH_LINKED_APPS = "GitOpsFetchLinkedApps";
+
+  String ELASTIGROUP_BG_STAGE_SETUP = "ElastigroupBGStageSetup";
+  String ELASTIGROUP_SWAP_ROUTE = "ElastigroupSwapRoute";
 }
```
[the remark this answers is not in the document]
Comment (arvind-choudhary-h) [reply 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java:89]: Yes, we just added independent objects for now. Second PR is on the row.

File: 970-ng-commons/src/main/java/io/harness/beans/SwaggerConstants.java (hunk 1 above)
Comment (arvind-choudhary-h) [reply 970-ng-commons/src/main/java/io/harness/beans/SwaggerConstants.java:41]: I see that both type of classpaths are provided in this Swagger. And it works!

**Review by lovish1999 — COMMENTED** [review]
LGTM

> Reviewed 53 of 53 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41492#-:-NJBAz1f-NgQLnAl0SqC:bb6ywmx)* status: 2 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained.

**Review by lovish1999 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41492)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by vitsafronovici — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41492)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]

**Pull request merged**